cicecore/cicedynB/general/ice_forcing.F90

```diff
@@ -5228,16 +5186,19 @@ subroutine get_wave_spec
          ! read more realistic data from a file
          if ((trim(wave_spec_type) == 'constant').OR.(trim(wave_spec_type) == 'random')) then
             if (trim(wave_spec_file(1:4)) == 'unkn') then
-               call abort_ice (subname//'ERROR: wave_spec_file '//trim(wave_spec_file))
+               call abort_ice (subname//'ERROR: wave_spec_file '//trim(wave_spec_file), &
+                  file=__FILE__, line=__LINE__)
             else
-#ifdef ncdf
+#ifdef USE_NETCDF
                call ice_open_nc(wave_spec_file,fid)
                call ice_read_nc_xyf (fid, 1, 'efreq', wave_spectrum(:,:,:,:), dbug, &
                                      field_loc_center, field_type_scalar)
                call ice_close_nc(fid)
 #else
-               write (nu_diag,*) "wave spectrum file not available, requires ncdf"
+               write (nu_diag,*) "wave spectrum file not available, requires cpp USE_NETCDF"
                write (nu_diag,*) "wave spectrum file not available, using default profile"
+               call abort_ice (subname//'ERROR: wave_spec_file '//trim(wave_spec_file), &
```

Comment:
As mentioned earlier I think that the abort should be in the ice_open_nc

Reply:
I did add the abort because I thought it was the right thing to do.  I think it's dangerous for a user to setup use of the wave spec file and then for it to relatively silently revert to the default values at runtime.  What should happen is the user should modify their namelist so it uses the default values and does not try to read a file.  That would give the same result but be explicit.  Another option would be to add a binary file reading capability for when netcdf is not available, but that's probably not the way to go at this time.  I would be interested in hearing what others think about this silent mode.  We have shifted the overall implementation from being one that changes the user settings at runtime to fix conflicts to one that aborts when there are conflicts.  I think this is a place where we want to follow that strategy.

Reply:
I agree that it is dangerous to revert to a default setting if the file is missing. This may happen from time to time in a operational setup due to delays of a required data flow. The abort option is better.

Reply:
That indeed makes sense. This is the kind of explanations that I think we should put in commit messages :) This way we can read the commit messages and look at the changes and understand faster why these changes are made.

Reply:
You're right, we should do a better of explaining changes in either our commit messages or the PR.  I'll try to do better.

Reply:
Just a note, I would prefer we use the commit messages; this way the information is encoded in the repo. If GitHub goes under someday for whatever reason, the commit messages stay, but the PR discussions might be more difficult to retrieve. I think I read somewhere (a very early PR?) that the GitHub projects for CICE and Icepack themselves are backed up somewhere (how regularly?), but this is not documented anywhere...

Reply:
I just added a couple sentences to the git workflow guide to indicate we want comprehensive commit messages.  I agree it's the right way to go.